--- a/Makefile.flags
+++ b/Makefile.flags
@@ -58,11 +58,6 @@ CFLAGS += $(call cc-option,-fno-builtin-
 CFLAGS += $(call cc-option,-fno-guess-branch-probability,)
 CFLAGS += $(call cc-option,-funsigned-char,)
 
-ifneq ($(CC),clang)
-# "clang-9: warning: argument unused during compilation: '-static-libgcc'"
-CFLAGS += $(call cc-option,-static-libgcc,)
-endif
-
 CFLAGS += $(call cc-option,-falign-functions=1,)
 ifneq ($(CC),clang)
 # "clang-9: warning: optimization flag '-falign-jumps=1' is not supported" (and same for other two)
