--- a/Makefile
+++ b/Makefile
@@ -54,7 +54,7 @@ CCOPTS = -O2 $(EXTRA_CCOPTS)
 WFLAGS := -Wall -Wstrict-prototypes  -Wmissing-prototypes
 WFLAGS += -Wmissing-declarations -Wold-style-definition -Wformat=2
 
-CFLAGS := $(WFLAGS) $(CCOPTS) -I../include -I../include/uapi $(DEFINES) $(CFLAGS)
+CFLAGS := $(WFLAGS) -I../include -I../include/uapi $(EXTRA_CCOPTS) $(DEFINES) $(CFLAGS)
 YACCFLAGS = -d -t -v
 
 SUBDIRS=lib ip tc bridge misc genl tipc devlink rdma man
