--- a/pppd/plugins/pptp/pptp_ctrl.c
+++ b/pppd/plugins/pptp/pptp_ctrl.c
@@ -16,6 +16,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <fcntl.h>
+#include "pppd.h"
 #include "pptp_msg.h"
 #include "pptp_ctrl.h"
 #include "pptp_options.h"
--- a/pppd/plugins/radius/radiusclient.h
+++ b/pppd/plugins/radius/radiusclient.h
@@ -440,6 +440,7 @@ UINT4 rc_get_ipaddr __P((char *));
 int rc_good_ipaddr __P((char *));
 const char *rc_ip_hostname __P((UINT4));
 UINT4 rc_own_ipaddress __P((void));
+UINT4 rc_own_bind_ipaddress __P((void));
 
 
 /*	sendserver.c		*/
--- a/pppd/pppd.h
+++ b/pppd/pppd.h
@@ -50,6 +50,8 @@
 #define __PPPD_H__
 
 #include <stdio.h>		/* for FILE */
+#include <stdlib.h>
+#include <unistd.h>
 #include <limits.h>		/* for NGROUPS_MAX */
 #include <sys/param.h>		/* for MAXPATHLEN and BSD4_4, if defined */
 #include <sys/types.h>		/* for u_int32_t, if defined */
