From 02f87543399ca311651dc446a830f0e24d21061c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20Men=C5=A1=C3=ADk?= <pemensik@redhat.com>
Date: Tue, 22 Nov 2022 22:51:11 +0000
Subject: [PATCH] fixup! Handle DS records for unsupported crypto algorithms.

---
 src/dnssec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dnssec.c b/src/dnssec.c
index ce01b2e..219ba9a 100644
--- a/src/dnssec.c
+++ b/src/dnssec.c
@@ -1068,7 +1068,7 @@ int dnssec_validate_ds(time_t now, struct dns_header *header, size_t plen, char
 	      algo = *p++;
 	      digest = *p++;
 	      
-	      if (!ds_digest_name(digest) || !ds_digest_name(digest))
+	      if (!ds_digest_name(digest) || !algo_digest_name(algo))
 		{
 		  a.log.keytag = keytag;
 		  a.log.algo = algo;
-- 
2.38.1

