From 20b4a4ea5b19f3143add1342dde74a0be2b30a86 Mon Sep 17 00:00:00 2001
From: Simon Kelley <simon@thekelleys.org.uk>
Date: Thu, 7 Jul 2022 20:56:07 +0100
Subject: [PATCH 19/20] Fix comment typo.

---
 src/hash-questions.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hash-questions.c b/src/hash-questions.c
index 4b1dd91..adcf62c 100644
--- a/src/hash-questions.c
+++ b/src/hash-questions.c
@@ -23,7 +23,7 @@
 
    The hash used is SHA-256. If we're building with DNSSEC support,
    we use the Nettle cypto library. If not, we prefer not to
-   add a dependency on Nettle, and use a stand-alone implementaion. 
+   add a dependency on Nettle, and use a stand-alone implementation. 
 */
 
 #include "dnsmasq.h"
-- 
2.37.1

