From 153eeb070b2135cac9e12fc38e944ca8683b74c7 Mon Sep 17 00:00:00 2001
From: Simon Kelley <simon@thekelleys.org.uk>
Date: Thu, 30 Mar 2023 16:00:04 +0100
Subject: [PATCH 19/41] Optimise no-action case in rrfilter().

---
 src/rrfilter.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/rrfilter.c b/src/rrfilter.c
index d98236e..d380e14 100644
--- a/src/rrfilter.c
+++ b/src/rrfilter.c
@@ -167,6 +167,9 @@ size_t rrfilter(struct dns_header *header, size_t *plen, int mode)
   size_t rr_found = 0;
   int i, rdlen, qtype, qclass, chop_an, chop_ns, chop_ar;
 
+  if (mode == RRFILTER_CONF && !daemon->filter_rr)
+    return 0;
+  
   if (ntohs(header->qdcount) != 1 ||
       !(p = skip_name(p, header, *plen, 4)))
     return 0;
-- 
2.41.0

