From 52e6ad2761e4e39c3e89fe3b8d68b98cf538a27c Mon Sep 17 00:00:00 2001
From: Simon Kelley <simon@thekelleys.org.uk>
Date: Mon, 17 Apr 2023 21:25:30 +0100
Subject: [PATCH 34/44] Log failure to determine MAC address in DHCPv6.

---
 src/rfc3315.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/rfc3315.c b/src/rfc3315.c
index bc09887..d513844 100644
--- a/src/rfc3315.c
+++ b/src/rfc3315.c
@@ -457,6 +457,8 @@ static int dhcp6_no_relay(struct state *state, int msg_type, unsigned char *inbu
 	    state->tags = &mac_opt->netid;
 	  }
     }
+  else if (option_bool(OPT_LOG_OPTS))
+    my_syslog(MS_DHCP | LOG_INFO, _("%u cannot determine client MAC address"), state->xid);
   
   if ((opt = opt6_find(state->packet_options, state->end, OPTION6_FQDN, 1)))
     {
-- 
2.42.0

