From 9bbf098a970c9e5fa251939208e25fb21064d1be Mon Sep 17 00:00:00 2001
From: Simon Kelley <simon@thekelleys.org.uk>
Date: Fri, 26 May 2023 18:19:15 +0100
Subject: [PATCH 41/44] =/== typo in last commit.

---
 src/forward.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/forward.c b/src/forward.c
index 9398353..95213c5 100644
--- a/src/forward.c
+++ b/src/forward.c
@@ -1961,7 +1961,7 @@ static ssize_t tcp_talk(int first, int last, int start, unsigned char *packet,
 	  
 	  if (errno == 0)
 	    data_sent = 1;
-	  else if (errno = ETIMEDOUT || errno == EHOSTUNREACH)
+	  else if (errno == ETIMEDOUT || errno == EHOSTUNREACH)
 	    timedout = 1;
 #endif
 	  
-- 
2.42.0

