--- a/src/ap/wpa_auth.c
+++ b/src/ap/wpa_auth.c
@@ -2732,7 +2732,7 @@ SM_STATE(WPA_PTK, PTKCALCNEGOTIATING)
 		idx = bitfield_get_first_zero(wpa_auth->ip_pool);
 		if (idx >= 0) {
 			u32 start = WPA_GET_BE32(wpa_auth->conf.ip_addr_start);
-			bitfield_set(wpa_auth->ip_pool, idx);
+			hostapd_bitfield_set(wpa_auth->ip_pool, idx);
 			WPA_PUT_BE32(sm->ip_addr, start + idx);
 			wpa_printf(MSG_DEBUG,
 				   "P2P: Assigned IP address %u.%u.%u.%u to "
--- a/src/utils/bitfield.c
+++ b/src/utils/bitfield.c
@@ -37,7 +37,7 @@ void bitfield_free(struct bitfield *bf)
 }
 
 
-void bitfield_set(struct bitfield *bf, size_t bit)
+void hostapd_bitfield_set(struct bitfield *bf, size_t bit)
 {
 	if (bit >= bf->max_bits)
 		return;
--- a/src/utils/bitfield.h
+++ b/src/utils/bitfield.h
@@ -13,7 +13,7 @@ struct bitfield;
 
 struct bitfield * bitfield_alloc(size_t max_bits);
 void bitfield_free(struct bitfield *bf);
-void bitfield_set(struct bitfield *bf, size_t bit);
+void hostapd_bitfield_set(struct bitfield *bf, size_t bit);
 void bitfield_clear(struct bitfield *bf, size_t bit);
 int bitfield_is_set(struct bitfield *bf, size_t bit);
 int bitfield_get_first_zero(struct bitfield *bf);
--- a/src/utils/utils_module_tests.c
+++ b/src/utils/utils_module_tests.c
@@ -141,7 +141,7 @@ static int bitfield_tests(void)
 			errors++;
 		if (i > 0 && bitfield_is_set(bf, i - 1))
 			errors++;
-		bitfield_set(bf, i);
+		hostapd_bitfield_set(bf, i);
 		if (!bitfield_is_set(bf, i))
 			errors++;
 		bitfield_clear(bf, i);
@@ -154,7 +154,7 @@ static int bitfield_tests(void)
 			errors++;
 		if (i > 0 && bitfield_is_set(bf, i - 1))
 			errors++;
-		bitfield_set(bf, i);
+		hostapd_bitfield_set(bf, i);
 		if (bitfield_is_set(bf, i))
 			errors++;
 		bitfield_clear(bf, i);
@@ -165,7 +165,7 @@ static int bitfield_tests(void)
 	for (i = 0; i < 123; i++) {
 		if (bitfield_is_set(bf, i) || bitfield_is_set(bf, i + 1))
 			errors++;
-		bitfield_set(bf, i);
+		hostapd_bitfield_set(bf, i);
 		if (!bitfield_is_set(bf, i))
 			errors++;
 	}
@@ -181,7 +181,7 @@ static int bitfield_tests(void)
 	for (i = 0; i < 123; i++) {
 		if (bitfield_get_first_zero(bf) != i)
 			errors++;
-		bitfield_set(bf, i);
+		hostapd_bitfield_set(bf, i);
 	}
 	if (bitfield_get_first_zero(bf) != -1)
 		errors++;
@@ -191,7 +191,7 @@ static int bitfield_tests(void)
 		bitfield_clear(bf, i);
 		if (bitfield_get_first_zero(bf) != i)
 			errors++;
-		bitfield_set(bf, i);
+		hostapd_bitfield_set(bf, i);
 	}
 	if (bitfield_get_first_zero(bf) != -1)
 		errors++;
@@ -204,7 +204,7 @@ static int bitfield_tests(void)
 	if (bitfield_get_first_zero(bf) != 0)
 		errors++;
 	for (i = 0; i < 8; i++)
-		bitfield_set(bf, i);
+		hostapd_bitfield_set(bf, i);
 	if (bitfield_get_first_zero(bf) != -1)
 		errors++;
 	bitfield_free(bf);
--- a/wpa_supplicant/rrm.c
+++ b/wpa_supplicant/rrm.c
@@ -1001,7 +1001,7 @@ static int wpas_rm_handle_beacon_req_sub
 		}
 
 		for (i = 0; i < slen; i++)
-			bitfield_set(data->eids, subelem[i]);
+			hostapd_bitfield_set(data->eids, subelem[i]);
 		break;
 	case WLAN_BEACON_REQUEST_SUBELEM_AP_CHANNEL:
 		/* Skip - it will be processed when freqs are added */
