--- a/src/utils.c
+++ b/src/utils.c
@@ -225,27 +225,6 @@ FATAL(const char *msg)
     exit(-1);
 }
 
-void *
-ss_malloc(size_t size)
-{
-    void *tmp = malloc(size);
-    if (tmp == NULL)
-        exit(EXIT_FAILURE);
-    return tmp;
-}
-
-void *
-ss_realloc(void *ptr, size_t new_size)
-{
-    void *new = realloc(ptr, new_size);
-    if (new == NULL) {
-        free(ptr);
-        ptr = NULL;
-        exit(EXIT_FAILURE);
-    }
-    return new;
-}
-
 void
 usage()
 {
--- a/src/utils.h
+++ b/src/utils.h
@@ -220,13 +220,31 @@ char *ss_strndup(const char *s, size_t n
 int set_nofile(int nofile);
 #endif
 
-void *ss_malloc(size_t size);
-void *ss_realloc(void *ptr, size_t new_size);
-
 #define ss_free(ptr)     \
     do {                 \
         free(ptr);       \
         ptr = NULL;      \
     } while (0)
 
+inline void *
+ss_malloc(size_t size)
+{
+    void *tmp = malloc(size);
+    if (tmp == NULL)
+        exit(EXIT_FAILURE);
+    return tmp;
+}
+
+inline void *
+ss_realloc(void *ptr, size_t new_size)
+{
+    void *new = realloc(ptr, new_size);
+    if (new == NULL) {
+        free(ptr);
+        ptr = NULL;
+        exit(EXIT_FAILURE);
+    }
+    return new;
+}
+
 #endif // _UTILS_H
