--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -111,10 +111,3 @@ ss_redir_CFLAGS = $(AM_CFLAGS) -DMODULE_
 ss_redir_LDADD = $(SS_COMMON_LIBS)
 ss_redir_LDADD += $(top_builddir)/libudns/libudns.la
 endif
-
-lib_LTLIBRARIES = libshadowsocks-libev.la
-libshadowsocks_libev_la_SOURCES = $(ss_local_SOURCES)
-libshadowsocks_libev_la_CFLAGS = $(ss_local_CFLAGS) -DLIB_ONLY
-libshadowsocks_libev_la_LDFLAGS = -version-info $(VERSION_INFO)
-libshadowsocks_libev_la_LIBADD = $(ss_local_LDADD)
-include_HEADERS = shadowsocks.h
