--- a/src/local.c
+++ b/src/local.c
@@ -1451,12 +1451,6 @@ main(int argc, char **argv)
         FATAL("failed to switch user");
     }
 
-#ifndef __MINGW32__
-    if (geteuid() == 0){
-        LOGI("running from root user");
-    }
-#endif
-
     // Disable retry
     keep_resolving = 0;
 
--- a/src/manager.c
+++ b/src/manager.c
@@ -852,12 +852,6 @@ main(int argc, char **argv)
         FATAL("failed to switch user");
     }
 
-#ifndef __MINGW32__
-    if (geteuid() == 0){
-        LOGI("running from root user");
-    }
-#endif
-
     struct passwd *pw   = getpwuid(getuid());
     const char *homedir = pw->pw_dir;
     working_dir_size = strlen(homedir) + 15;
--- a/src/redir.c
+++ b/src/redir.c
@@ -1048,10 +1048,6 @@ main(int argc, char **argv)
         FATAL("failed to switch user");
     }
 
-    if (geteuid() == 0){
-        LOGI("running from root user");
-    }
-
     ev_run(loop, 0);
 
     return 0;
--- a/src/server.c
+++ b/src/server.c
@@ -1890,9 +1890,7 @@ main(int argc, char **argv)
     }
 
 #ifndef __MINGW32__
-    if (geteuid() == 0){
-        LOGI("running from root user");
-    } else if (firewall) {
+    if (geteuid() != 0 && firewall) {
         LOGE("firewall setup requires running from root user");
         exit(-1);
     }
--- a/src/tunnel.c
+++ b/src/tunnel.c
@@ -1041,12 +1041,6 @@ main(int argc, char **argv)
         FATAL("failed to switch user");
     }
 
-#ifndef __MINGW32__
-    if (geteuid() == 0){
-        LOGI("running from root user");
-    }
-#endif
-
     ev_run(loop, 0);
 
 #ifdef __MINGW32__
