--- a/src/local.c
+++ b/src/local.c
@@ -1563,12 +1563,6 @@ main(int argc, char **argv)
         FATAL("failed to switch user");
     }
 
-#ifndef __MINGW32__
-    if (geteuid() == 0) {
-        LOGI("running from root user");
-    }
-#endif
-
     // Init connections
     cork_dllist_init(&connections);
 
--- a/src/manager.c
+++ b/src/manager.c
@@ -1137,12 +1137,6 @@ main(int argc, char **argv)
         FATAL("failed to switch user");
     }
 
-#ifndef __MINGW32__
-    if (geteuid() == 0) {
-        LOGI("running from root user");
-    }
-#endif
-
     struct passwd *pw   = getpwuid(getuid());
     const char *homedir = pw->pw_dir;
     working_dir_size = strlen(homedir) + 15;
--- a/src/redir.c
+++ b/src/redir.c
@@ -1110,10 +1110,6 @@ main(int argc, char **argv)
         FATAL("failed to switch user");
     }
 
-    if (geteuid() == 0) {
-        LOGI("running from root user");
-    }
-
     ev_run(loop, 0);
 
     return 0;
--- a/src/server.c
+++ b/src/server.c
@@ -1953,9 +1953,7 @@ main(int argc, char **argv)
     }
 
 #ifndef __MINGW32__
-    if (geteuid() == 0) {
-        LOGI("running from root user");
-    } else if (firewall) {
+    if (geteuid() != 0 && firewall) {
         LOGE("firewall setup requires running from root user");
         exit(-1);
     }
--- a/src/tunnel.c
+++ b/src/tunnel.c
@@ -1106,12 +1106,6 @@ main(int argc, char **argv)
         FATAL("failed to switch user");
     }
 
-#ifndef __MINGW32__
-    if (geteuid() == 0) {
-        LOGI("running from root user");
-    }
-#endif
-
     ev_run(loop, 0);
 
 #ifdef __MINGW32__
