--- a/Makefile.am
+++ b/Makefile.am
@@ -1,5 +1,5 @@
 if USE_SYSTEM_SHARED_LIB
-SUBDIRS = libcork libipset src
+SUBDIRS = libcork libipset libudns src
 else
 SUBDIRS = libsodium libcork libipset libudns libev src
 endif
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -9,10 +9,10 @@ endif
 AM_CFLAGS += $(PTHREAD_CFLAGS)
 if !USE_SYSTEM_SHARED_LIB
 AM_CFLAGS += -I$(top_srcdir)/libev
-AM_CFLAGS += -I$(top_srcdir)/libudns
 AM_CFLAGS += -I$(top_srcdir)/libsodium/src/libsodium/include
 endif
 AM_CFLAGS += -I$(top_srcdir)/libipset/include
+AM_CFLAGS += -I$(top_srcdir)/libudns
 AM_CFLAGS += -I$(top_srcdir)/libcork/include
 AM_CFLAGS += $(LIBPCRE_CFLAGS)
 
@@ -86,17 +86,10 @@ fs_local_LDADD = $(FS_COMMON_LIBS)
 fs_tunnel_LDADD = $(FS_COMMON_LIBS)
 fs_server_LDADD = $(FS_COMMON_LIBS)
 fs_manager_LDADD = $(FS_COMMON_LIBS)
-if USE_SYSTEM_SHARED_LIB
-fs_local_LDADD += -ludns
-fs_tunnel_LDADD += -ludns
-fs_server_LDADD += -ludns
-fs_manager_LDADD += -ludns
-else
 fs_local_LDADD += $(top_builddir)/libudns/libudns.la
 fs_tunnel_LDADD += $(top_builddir)/libudns/libudns.la
 fs_server_LDADD += $(top_builddir)/libudns/libudns.la
 fs_manager_LDADD += $(top_builddir)/libudns/libudns.la
-endif
 
 fs_local_CFLAGS = $(AM_CFLAGS) -DMODULE_LOCAL
 fs_tunnel_CFLAGS = $(AM_CFLAGS) -DMODULE_TUNNEL
@@ -122,9 +115,5 @@ fs_redir_SOURCES = utils.c \
 
 fs_redir_CFLAGS = $(AM_CFLAGS) -DMODULE_REDIR
 fs_redir_LDADD = $(FS_COMMON_LIBS)
-if USE_SYSTEM_SHARED_LIB
-fs_redir_LDADD += -ludns
-else
 fs_redir_LDADD += $(top_builddir)/libudns/libudns.la
 endif
-endif
--- a/configure.ac
+++ b/configure.ac
@@ -300,13 +300,8 @@ AM_COND_IF([USE_SYSTEM_SHARED_LIB],[
 AC_CONFIG_FILES([ Makefile
                  libcork/Makefile
                  libipset/Makefile
+                 libudns/Makefile
                  src/Makefile])
-AM_COND_IF([USE_SYSTEM_SHARED_LIB],[
-    AC_CHECK_LIB([udns], [dns_dnlen], ,[AC_MSG_ERROR([Couldn't find libudns. Try installing libudns-dev or udns-devel.])])
-    AC_CHECK_LIB([ev], [ev_loop_destroy], ,[AC_MSG_ERROR([Couldn't find libev. Try installing libev-dev@<:@el@:>@.])])
-  ],
-  [AC_CONFIG_FILES([libudns/Makefile
-                 libev/Makefile])])
 
 AM_COND_IF([ENABLE_DOCUMENTATION],
   [AC_CONFIG_FILES([doc/Makefile])
