--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -9,11 +9,11 @@ endif
 AM_CFLAGS += $(PTHREAD_CFLAGS)
 if !USE_SYSTEM_SHARED_LIB
 AM_CFLAGS += -I$(top_srcdir)/libev
-AM_CFLAGS += -I$(top_srcdir)/libcares
 AM_CFLAGS += -I$(top_srcdir)/libsodium/src/libsodium/include
 endif
 AM_CFLAGS += -I$(top_srcdir)/libipset/include
 AM_CFLAGS += -I$(top_srcdir)/libcork/include
+AM_CFLAGS += -I$(top_srcdir)/libcares
 AM_CFLAGS += $(LIBPCRE_CFLAGS)
 
 FS_COMMON_LIBS = $(top_builddir)/libipset/libipset.la \
@@ -88,17 +88,10 @@ fs_local_LDADD = $(FS_COMMON_LIBS)
 fs_tunnel_LDADD = $(FS_COMMON_LIBS)
 fs_server_LDADD = $(FS_COMMON_LIBS)
 fs_manager_LDADD = $(FS_COMMON_LIBS)
-if USE_SYSTEM_SHARED_LIB
-fs_local_LDADD += -lcares
-fs_tunnel_LDADD += -lcares
-fs_server_LDADD += -lcares
-fs_manager_LDADD += -lcares
-else
 fs_local_LDADD += $(top_builddir)/libcares/libcares.la
 fs_tunnel_LDADD += $(top_builddir)/libcares/libcares.la
 fs_server_LDADD += $(top_builddir)/libcares/libcares.la
 fs_manager_LDADD += $(top_builddir)/libcares/libcares.la
-endif
 
 fs_local_CFLAGS = $(AM_CFLAGS) -DMODULE_LOCAL
 fs_tunnel_CFLAGS = $(AM_CFLAGS) -DMODULE_TUNNEL
@@ -124,9 +117,5 @@ fs_redir_SOURCES = utils.c \
 
 fs_redir_CFLAGS = $(AM_CFLAGS) -DMODULE_REDIR
 fs_redir_LDADD = $(FS_COMMON_LIBS)
-if USE_SYSTEM_SHARED_LIB
-fs_redir_LDADD += -lcares
-else
 fs_redir_LDADD += $(top_builddir)/libcares/libcares.la
 endif
-endif
--- a/configure.ac
+++ b/configure.ac
@@ -286,30 +286,13 @@ AC_CHECK_LIB(socket, connect)
 dnl Checks for library functions.
 AC_CHECK_FUNCS([malloc memset posix_memalign socket])
 
-AM_COND_IF([USE_SYSTEM_SHARED_LIB],[
-    dnl Check c-ares header
-    AC_CHECK_HEADERS([ares.h], [], [AC_MSG_ERROR([Couldn't find libcares. Try installing libc-ares-dev or c-ares-devel.])])
-],[])
-
-AM_COND_IF([USE_SYSTEM_SHARED_LIB],[
-    AC_CHECK_LIB([sodium], [sodium_init], ,[
-       AC_MSG_ERROR([Couldn't find libsodium. Try installing libsodium-dev@<:@el@:>@.])
-    ])
-  ],
-  [AC_CONFIG_SUBDIRS([libsodium])
-  AC_CONFIG_SUBDIRS([libcares])
-  ])
+AC_CONFIG_SUBDIRS([libcares])
 
 AC_CONFIG_FILES([ Makefile
                  libcork/Makefile
                  libipset/Makefile
+                 libcares/Makefile
                  src/Makefile])
-AM_COND_IF([USE_SYSTEM_SHARED_LIB],[
-    AC_CHECK_LIB([cares], [ares_library_init], [LIBS="-lcares $LIBS"], [AC_MSG_ERROR([Couldn't find libc-ares. Try installing libc-ares-dev or c-ares-devel.])])
-    AC_CHECK_LIB([ev], [ev_loop_destroy], ,[AC_MSG_ERROR([Couldn't find libev. Try installing libev-dev@<:@el@:>@.])])
-  ],
-  [AC_CONFIG_FILES([libcares/Makefile
-                 libev/Makefile])])
 
 AM_COND_IF([ENABLE_DOCUMENTATION],
   [AC_CONFIG_FILES([doc/Makefile])
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,5 +1,5 @@
 if USE_SYSTEM_SHARED_LIB
-SUBDIRS = libcork libipset src
+SUBDIRS = libcork libipset libcares src
 else
 SUBDIRS = libsodium libcork libipset libcares libev src
 endif
