--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -3,7 +3,7 @@ VERSION_INFO = 2:0:0
 if ENABLE_OPTIMIZATION
 AM_CFLAGS = -O3 -pipe -ffast-math -ftree-vectorize -Wall -Werror -fno-strict-aliasing -std=gnu99 -D_GNU_SOURCE
 else
-AM_CFLAGS = -Os -Wall -Werror -fno-strict-aliasing -std=gnu99 -D_GNU_SOURCE
+AM_CFLAGS = -Wall -Werror -fno-strict-aliasing -std=gnu99 -D_GNU_SOURCE
 endif
 
 AM_CFLAGS += $(PTHREAD_CFLAGS)
