--- a/libcares/m4/xc-val-flgs.m4
+++ b/libcares/m4/xc-val-flgs.m4
@@ -33,7 +33,6 @@ AC_DEFUN([_XC_CHECK_VAR_LIBS], [
         :
         ;;
       *)
-        xc_bad_var_libs=yes
         ;;
     esac
   done
@@ -53,16 +52,12 @@ AC_DEFUN([_XC_CHECK_VAR_LDFLAGS], [
   for xc_word in $LDFLAGS; do
     case "$xc_word" in
       -D*)
-        xc_bad_var_ldflags=yes
         ;;
       -U*)
-        xc_bad_var_ldflags=yes
         ;;
       -I*)
-        xc_bad_var_ldflags=yes
         ;;
       -l* | --library=*)
-        xc_bad_var_ldflags=yes
         ;;
     esac
   done
@@ -98,13 +93,10 @@ AC_DEFUN([_XC_CHECK_VAR_CPPFLAGS], [
   for xc_word in $CPPFLAGS; do
     case "$xc_word" in
       -rpath*)
-        xc_bad_var_cppflags=yes
         ;;
       -L* | --library-path=*)
-        xc_bad_var_cppflags=yes
         ;;
       -l* | --library=*)
-        xc_bad_var_cppflags=yes
         ;;
     esac
   done
@@ -137,22 +129,16 @@ AC_DEFUN([_XC_CHECK_VAR_CFLAGS], [
   for xc_word in $CFLAGS; do
     case "$xc_word" in
       -D*)
-        xc_bad_var_cflags=yes
         ;;
       -U*)
-        xc_bad_var_cflags=yes
         ;;
       -I*)
-        xc_bad_var_cflags=yes
         ;;
       -rpath*)
-        xc_bad_var_cflags=yes
         ;;
       -L* | --library-path=*)
-        xc_bad_var_cflags=yes
         ;;
       -l* | --library=*)
-        xc_bad_var_cflags=yes
         ;;
     esac
   done
